lib/ansible/modules/cloud/amazon/cloudwatchlogs_log_group.py

```diff
@@ -0,0 +1,298 @@
+    try:
+        permited_values = [1, 3, 5, 7, 14, 30, 60, 90, 120, 150, 180, 365, 400, 545, 731, 1827, 3653]
+
+        if retention in permited_values:
+            response = client.put_retention_policy(logGroupName=log_group_name,
+                                                   retentionInDays=retention)
+        else:
+            delete_log_group(client=client, log_group_name=log_group_name, module=module)
+            module.fail_json(msg="Invalid retention value. Valid values are: [1, 3, 5, 7, 14, 30, 60, 90, 120, 150, 180, 365, 400, 545, 731, 1827, 3653]")
+    except botocore.exceptions.ClientError as e:
+        module.fail_json(msg=e.response, exception=traceback.format_exc(),
+                         **camel_dict_to_snake_dict(e.response))
+
+
+def delete_log_group(client, log_group_name, module):
+    try:
+        desc_log_group = describe_log_group(client=client,
+                                            log_group_name=log_group_name,
+                                            module=module)
+
+        if 'logGroups' in desc_log_group:
+            for i in desc_log_group['logGroups']:
+                if log_group_name == i['logGroupName']:
+                    client.delete_log_group(logGroupName=log_group_name)
+
+    except botocore.exceptions.ClientError as e:
+        module.fail_json(msg=e.response, exception=traceback.format_exc(),
+                         **camel_dict_to_snake_dict(e.response))
+
+
+def describe_log_group(client, log_group_name, module):
+    try:
+        desc_log_group = client.describe_log_groups(logGroupNamePrefix=log_group_name)
+        return desc_log_group
+    except botocore.exceptions.ClientError as e:
+        module.fail_json(msg=e.response, exception=traceback.format_exc(),
+                         **camel_dict_to_snake_dict(e.response))
+
+
+def main():
+    argument_spec = ec2_argument_spec()
+    argument_spec.update(dict(
+        log_group_name=dict(required=True, type='str'),
+        state=dict(choices=['present', 'absent', 'get'],
+                   default='present'),
+        kms_key_id=dict(required=False, type='str'),
+        tags=dict(required=False, type='dict'),
+        retention=dict(required=False, type='int'),
+        overwrite=dict(required=False, type='bool', default=False)
+    ))
+
+    module = AnsibleModule(argument_spec=argument_spec, supports_check_mode=True)
+
+    if not HAS_BOTO3:
+        module.fail_json(msg='boto3 is required.')
+
+    try:
+        region, ec2_url, aws_connect_kwargs = get_aws_connection_info(module, boto3=True)
+        if not region:
+            module.fail_json(msg="Region must be specified as a parameter, in EC2_REGION or AWS_REGION environment variables or in boto configuration file")
+        logs = boto3_conn(module, conn_type='client', resource='logs', region=region, endpoint=ec2_url, **aws_connect_kwargs)
+    except botocore.exceptions.ProfileNotFound as e:
```

Comment:
You don't need to wrap the connection in try/except, it is now handled in module_utils/ec2 (https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/cloud/amazon/GUIDELINES.md#boto3-1).